compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/service/validator/InterceptorValidator.java

```diff
@@ -0,0 +1,98 @@
+import static io.ballerina.stdlib.graphql.compiler.service.validator.ValidatorUtils.INTERCEPTOR_EXECUTE;
+import static io.ballerina.stdlib.graphql.compiler.service.validator.ValidatorUtils.updateContext;
+
+/**
+ * Validate Interceptors in Ballerina GraphQL services.
+ */
+public class InterceptorValidator {
+
+    private final ClassDefinitionNode classDefinitionNode;
+    private final SyntaxNodeAnalysisContext context;
+    private boolean errorOccurred;
+
+    public InterceptorValidator(SyntaxNodeAnalysisContext context, ClassDefinitionNode classDefinitionNode) {
+        this.context = context;
+        this.classDefinitionNode = classDefinitionNode;
+        this.errorOccurred = false;
+    }
+
+    public void validate() {
+        NodeList<Node> members = this.classDefinitionNode.members();
+        for (Node member : members) {
+            if (member.kind() == SyntaxKind.TYPE_REFERENCE) {
+                if (isInterceptor(member)) {
+                    validateInterceptorService(members);
+                    break;
+                }
+            }
+        }
+    }
+
+    public boolean isErrorOccurred() {
+        return this.errorOccurred;
+    }
+
+    private void validateInterceptorService(NodeList<Node> members) {
+        for (Node node : members) {
+            validateInterceptorServiceMember(node);
+        }
+    }
+
+    private void validateInterceptorServiceMember(Node node) {
+        if (this.context.semanticModel().symbol(node).isEmpty()) {
+            return;
+        }
+        Symbol symbol = this.context.semanticModel().symbol(node).get();
+        Location location = node.location();
+        if (symbol.kind() == SymbolKind.METHOD) {
+            MethodSymbol methodSymbol = (MethodSymbol) symbol;
+            if (isRemoteMethod(methodSymbol)) {
+                validateRemoteMethod(methodSymbol, location);
+            }
+        } else if (symbol.kind() == SymbolKind.RESOURCE_METHOD) {
+            addDiagnostic(CompilationError.RESOURCE_METHOD_INSIDE_INTERCEPTOR, location);
+        }
+    }
+
+    private void validateRemoteMethod(MethodSymbol methodSymbol, Location location) {
+        if (methodSymbol.getName().isEmpty()) {
+            return;
+        }
+        if (!methodSymbol.getName().get().equals(INTERCEPTOR_EXECUTE)) {
+            addDiagnostic(CompilationError.INVALID_REMOTE_METHOD_INSIDE_INTERCEPTOR, location);
+        }
+    }
+
+    public boolean isInterceptor(Node member) {
+        if (this.context.semanticModel().symbol(member).isEmpty()) {
+            return false;
+        }
+        Symbol symbol = this.context.semanticModel().symbol(member).get();
+        TypeReferenceTypeSymbol typeReferenceTypeSymbol = (TypeReferenceTypeSymbol) symbol;
+        if (typeReferenceTypeSymbol.getName().isEmpty()) {
+            return false;
+        }
+        return GRAPHQL_INTERCEPTOR.equals(typeReferenceTypeSymbol.getName().get());
```

Comment:
This does not check the package name and the org, right?

Reply:
Missed that. Fixed in https://github.com/ballerina-platform/module-ballerina-graphql/pull/812/commits/0ccd32090205c2201acc05c687830c44b0ce4a2a